src/hotspot/share/runtime/thread.cpp

```diff
@@ -2205,22 +2210,26 @@ const char* JavaThread::name() const  {
 // descriptive string if there is no set name.
 const char* JavaThread::get_thread_name_string(char* buf, int buflen) const {
   const char* name_str;
-  oop thread_obj = threadObj();
-  if (thread_obj != NULL) {
-    oop name = java_lang_Thread::name(thread_obj);
-    if (name != NULL) {
-      if (buf == NULL) {
-        name_str = java_lang_String::as_utf8_string(name);
+  if (is_oop_safe()) {
```

Comment:
Same comment - it is the current thread that must be oop_safe. If the target thread has exited then we will detect that via the null threadObj.

Reply:
Agreed.